### PR TITLE
Python bp

### PR DESCRIPTION
**Pull request by thegentlemanphysicist**
here it is

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
```
Comment (gyanesh-mishra) [inline build.sh:1]: Not sure we need this

File: docker-compose.yml [hunk 2]
```diff
@@ -15,7 +15,47 @@ services:
     networks:
       - default
 
+  ####################### Database and API Definition #######################
+  db:
+    image: example-db-img
```
Comment (gyanesh-mishra) [inline docker-compose.yml:20]: the image directive is probably not doing much

File: docker-compose.yml [hunk 3]
```diff
@@ -15,7 +15,47 @@ services:
...
+    container_name: example-db
+    build:
+      context: ./services/app_api/main/db
+      dockerfile: Dockerfile
+    ports:
+      - 5435:5432
+    environment:
+      - POSTGRES_USER=postgres
+      - POSTGRES_PASSWORD=postgres
+    networks:
```
Comment (gyanesh-mishra) [inline docker-compose.yml:30]: There's already a default network defined. You can get rid of local-network

File: .gitignore [hunk 4]
```diff
@@ -9,6 +9,7 @@ dist
 *.swp
 .tern-port
 .nvimrc
+*.pyc
```
Comment (gyanesh-mishra) [inline .gitignore:14]: Add the .ds_store files here as well

File: services/app_api/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,26 @@
+# Base image
+FROM python:3.6.5-alpine
+
+# Install dependencies
+RUN apk update && \
```
Comment (gyanesh-mishra) [inline services/app_api/Dockerfile]: Do you need all these build tools?

File: services/app_api/entrypoint.sh [hunk 6]
```diff
@@ -0,0 +1,9 @@
+#!/bin/sh
+
+echo "Waiting for postgres..."
+while ! nc -z db 5432; do
+  sleep 0.1
+done
+echo "PostgreSQL started"
+
+python manage.py run -h 0.0.0.0
```
Comment (gyanesh-mishra) [inline services/app_api/entrypoint.sh]: Is this for migrations?

File: services/app_api/main/__init__.py [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+def create_app(script_info=None):
+    # Instantiate the app
+    app = Flask(__name__)
+
+    # Enable CORS
+    CORS(app)
+
+    # Get config
+    app_settings = os.getenv('APP_SETTINGS')
```
Comment (gyanesh-mishra) [inline services/app_api/main/__init__.py]: Can we move these to some config class?

File: services/app_api/main/api/models.py [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+    def __init__(self, string_field):
+        self.string_field = string_field
+
+    def to_json(self):
+        return {
+            'id': self.id,
+            'name': self.string_field,
+        }
+
+# ADD MODELS AS NEEDED, SEISM PROJECT HAS A USER MODEL DEFINED THAT MAY BE USEFULL
```
Comment (gyanesh-mishra) [inline services/app_api/main/api/models.py]: *Useful

File: services/app_api/main/db/create.sql [hunk 9]
```diff
@@ -0,0 +1,3 @@
+CREATE DATABASE db_prod;
```
Comment (gyanesh-mishra) [inline services/app_api/main/db/create.sql]: Is this needed?

File: services/app_api/swagger.yaml [hunk 10]
```diff
@@ -0,0 +1,334 @@
+openapi: 3.0.0
```
Comment (gyanesh-mishra) [inline services/app_api/swagger.yaml]: Is this needed?

File: services/app_api/test/test_example.py [hunk 11]
```diff
@@ -0,0 +1,127 @@
...
+            # self.assertIn('Norway rat', data['species'][0]['name'])
+            # self.assertIn('Rattus norvegicus', data['species'][0]['latin'])
+            # self.assertIn('Brown or dark grey in colour.',
+            #               data['species'][0]['description'])
+            # self.assertIn('1880-01-01', data['species'][0]['introduction'])
+            # self.assertIn(category.name, data['species'][0]['category'])
+
+
+    # # Basic happy path, ensure the /species route behaves correctly for POST
+    # def test_species_creation(self):
```
Comment (gyanesh-mishra) [inline services/app_api/test/test_example.py]: Same here

File: services/nginx/Dockerfile [hunk 12]
```diff
@@ -0,0 +1,9 @@
+FROM nginx:1.15.0-alpine
```
Comment (gyanesh-mishra) [inline services/nginx/Dockerfile]: Do we need nginx as well?

File: setup_api.sh [hunk 13]
```diff
@@ -0,0 +1,2 @@
+#!/bin/bash
```
Comment (gyanesh-mishra) [inline setup_api.sh:1]: Don't think this is needed as well

**Review by gyanesh-mishra — CHANGES_REQUESTED** [review]
Lots of un-necessary bloated stuff for a boilerplate IMO.

Also in conjunction to `client` directory can this be named `server`?

File: .gitignore (hunk 4 above)
Comment (thegentlemanphysicist) [reply .gitignore:14]: removed

File: services/app_api/test/test_example.py (hunk 11 above)
Comment (thegentlemanphysicist) [reply services/app_api/test/test_example.py]: removed the species stuff still need to work on tests

File: services/nginx/Dockerfile (hunk 12 above)
Comment (thegentlemanphysicist) [reply services/nginx/Dockerfile]: nginx and references to it are now removed

File: setup_api.sh (hunk 13 above)
Comment (thegentlemanphysicist) [reply setup_api.sh:1]: If I want to get the api running I needed this script run.  I added that content to the README at the base of the project

File: build.sh (hunk 1 above)
Comment (thegentlemanphysicist) [reply build.sh:1]: I've been running this to do api development locally, see Readme

File: services/app_api/main/db/create.sql (hunk 9 above)
Comment (thegentlemanphysicist) [reply services/app_api/main/db/create.sql]: I do not know

File: services/app_api/Dockerfile (hunk 5 above)
Comment (thegentlemanphysicist) [reply services/app_api/Dockerfile]: I do not know

File: docker-compose.yml (hunk 2 above)
Comment (thegentlemanphysicist) [reply docker-compose.yml:20]: I will trust your wisdom on docker compose file choices, but I'm not confident in my ability to not break things

File: services/app_api/swagger.yaml (hunk 10 above)
Comment (thegentlemanphysicist) [reply services/app_api/swagger.yaml]: This is my last todo.  It is up to the code challenge peeps whether or not they want to implement swagger in the boiler plate

**Pull request closed without merging**